connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerConfig.java

```diff
@@ -366,6 +369,29 @@ public WorkerConfig(ConfigDef definition, Map<String, String> props) {
         super(definition, props);
         logInternalConverterRemovalWarnings(props);
         logPluginPathConfigProviderWarning(props);
+        ignoreSubConfigs();
+    }
+
+    private void ignoreSubConfigs() {
+        subConfigPrefixes().forEach(this::ignoreAllWithPrefixes);
+        Arrays.asList(
+                KEY_CONVERTER_CLASS_CONFIG, VALUE_CONVERTER_CLASS_CONFIG, HEADER_CONVERTER_CLASS_CONFIG
+        ).forEach(this::ignore);
+    }
+
+    protected List<String> subConfigPrefixes() {
+        return new ArrayList<>(Arrays.asList(
+                KEY_CONVERTER_CLASS_CONFIG + ".",
+                VALUE_CONVERTER_CLASS_CONFIG + ".",
+                HEADER_CONVERTER_CLASS_CONFIG + ".",
```

Comment:
We could potentially replace this with actual instantiation and configuration of the key, value, and header converters specified in the worker config, but that may be wasteful of resources (especially since the `Converter` interface doesn't extend `Closeable` yet) and it's unclear how we'd want to handle failures encountered during that process (aborting worker startup is not an option as converter instantiation may fail due to transient errors).